tests/unit/test_config_ports.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Feature: Opentelemetry-collector exposes ports dynamically.
+
+Scenario: Add/remove a port to/from the config
+    When the otelcol config has a port defined in the config
+    Then the ports exposed on the charm unit should contain only those in the config
+"""
```

Comment:
This is great for itests because we practice one-pyfile-per-test, but for utests we don't have that restriction.
You already have "[loose-bdd](https://discourse.charmhub.io/t/practicing-loose-bdd-in-charm-unit-tests/6957)" comments, so if a module docstring feels too forced, feel free to drop it, or only keep the "feature" statement, or move it to a class docstring if you'd like to [group tests under a class](https://stackoverflow.com/questions/50016862/grouping-tests-in-pytest-classes-vs-plain-functions).